tools/python/runfiles/runfiles.py

```diff
@@ -161,6 +165,51 @@ def EnvVars(self):
     """
     return self._strategy.EnvVars()
 
+  def CurrentRepository(self, frame = 1):
+    """Returns the canonical name of the caller's Bazel repository.
+
+    This function uses the 'inspect' package to obtain the file containing the
+    frame-th caller of this method and returns the canonical name of the Bazel
+    repository containing that file.
+
+    For example, this function returns '' (the empty string) when called from
+    the main repository and 'bazel_tools' when called from the @bazel_tools
+    repository.
+
+    Args:
+      frame: int; the stack frame to return the repository name for. Defaults to
+       1, the caller of the CurrentRepository function.
+    Returns:
+      the canonical name of the Bazel repository containing the file containing
+      the frame-th caller of this function
+    Raises:
+      ValueError: if the caller cannot be determined or the caller's file path
+       is not contained in the Python runfiles tree
+    """
+    caller_path = inspect.getfile(sys._getframe(frame))
+    caller_runfiles_path = os.path.relpath(caller_path, self._python_runfiles_root)
+    if caller_runfiles_path.startswith(".." + os.path.sep):
+      raise ValueError('{} does not lie under the runfiles root {}'.format(caller_path, self._python_runfiles_root))
+
+    caller_runfiles_directory = caller_runfiles_path[:caller_runfiles_path.find(os.path.sep)]
+    if caller_runfiles_directory == MAIN_REPOSITORY_RUNFILES_DIRECTORY:
+      # The canonical name of the main repository (also know as the workspace)
+      # is the empty string.
+      return ''
+    # For all other repositories, the name of the runfiles directory is the
```

Comment:
Won't this return e.g. "bazel_tools~1.0"? Because that's the name of the directory in runfiles. The docstring says it should instead return e.g. "bazel_tools".